src/docker/dockerclient.cc

```diff
@@ -57,6 +57,82 @@ std::tuple<bool, std::string> DockerClient::getContainerState(const Json::Value&
   return {false, ""};
 }
 
+std::string DockerClient::getContainerHealth(const std::string& id) {
+  const std::string cmd{"http://localhost/containers/" + id + "/json"};
+  auto resp = http_client_->get(cmd, HttpInterface::kNoLimit);
+  if (!resp.isOk()) {
+    throw std::runtime_error("Request to dockerd has failed: " + cmd);
+  }
+  const auto container_info{resp.getJson()};
+  return container_info["State"]["Health"]["Status"].asString();
+}
+
+int DockerClient::getContainerExitCode(const std::string& id) {
+  const std::string cmd{"http://localhost/containers/" + id + "/json"};
+  auto resp = http_client_->get(cmd, HttpInterface::kNoLimit);
+  if (!resp.isOk()) {
+    throw std::runtime_error("Request to dockerd has failed: " + cmd);
+  }
+  const auto container_info{resp.getJson()};
+  return container_info["State"]["ExitCode"].asInt();
+}
```

Comment:
i know its a draft, but it would be nice if we could make that docker engine api call once and get both values.

Reply:
We don't both, either health or exit code is requested.